setup-gcloud/src/setup-gcloud.ts

```diff
@@ -49,15 +49,27 @@ async function run() {
     }
 
     // write the service account key to a temporary file
-    const tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
+    let tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
```

Comment:
I'd rather not overload the purpose of this variable. This variable is distinctly meant to represent a temporary file. In the case where a use wants a persistent file, we should use a separate variable to represent that.

Reply:
yes but later this variable is being used :
https://github.com/GoogleCloudPlatform/github-actions/blob/67a97a91f262da1798e5d0ec09b38057e89baf32/setup-gcloud/src/setup-gcloud.ts#L69
and
https://github.com/GoogleCloudPlatform/github-actions/blob/67a97a91f262da1798e5d0ec09b38057e89baf32/setup-gcloud/src/setup-gcloud.ts#L76

and I don't want to duplicate this logic for 2 different variables.
Also, IMO both representing the same thing, a file path, just it is being defined by 2 options: random or by user input

Reply:
Fair point :)